src/app/server/Dnssd.cpp

```diff
@@ -213,6 +221,9 @@ CHIP_ERROR DnssdServer::AdvertiseOperational()
         AddICDKeyToAdvertisement(advertiseParameters);
 #endif
 
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT
+        AddTCPSupportToAdvertisement(advertiseParameters);
```

Comment:
I don't get it.  If we just do this in the one spot, why do we need the complexity with the template and whatnot?  Or for that matter why do we need the separate function if it's a one-liner?

If you just copied what ICD does, note that ICD is calling AddICDKeyToAdvertisement from multiple places with different parameter types, and the logic for ICD is in fact complicated enough to warrant a separate function.